lnwallet/channel.go

```diff
@@ -5490,7 +5480,10 @@ func newIncomingHtlcResolution(signer input.Signer, localChanCfg *channeldb.Chan
 	}
 
 	// Next, we'll construct the full witness needed to satisfy the input
-	// of the success transaction.
+	// of the success transaction. We'll populate the witness stack with a
+	// blank preimage, which must be overwritten using ApplyPreimage with
+	// the discovered preimage before the witness can be valid.
+	var preimage [32]byte
 	successWitness, err := input.ReceiverHtlcSpendRedeem(
 		htlc.Signature, preimage[:], signer, &successSignDesc, successTx,
```

Comment:
Maybe not for this PR, but could eventually this whole witness generation be moved to the resolvers? Not to create a template here and fill in the preimage later?

Reply:
indeed, i'm not sure how best to accomplish that atm, but i added a todo

Reply:
added todo